docs/manual-EN/3.build-develop-and-administration/1.build/1.build-source-code.md

```diff
@@ -1,244 +1,200 @@
-# Build Source Code
+# Build From Source
+## Caveats
+We have tested building on various environments, from CentOS 6 to 8, Ubuntu 16.04 to 19.04, Fedora 28 to 30, with GCC from 7.1.0 to 9.2.0 and recent Clang++, even with the devtoolset of Red Hat and CentOS. But due to the complexity of building environments, we still cannot guarantee that we have considered for all kinds of situations. If any problem encountered, please fire an issue or open a pull request to let us know.
```

Comment:
We have tested building on various environments, including CentOS 6 to 8, Ubuntu 16.04 to 19.04, Fedora 28 to 30, GCC 7.1.0 to 9.2.0 and latest Clang++, even the devtoolset of Red Hat and CentOS.

Reply:
...we have covered all kinds of situations